server/text_generation_server/models/causal_lm.py

```diff
@@ -922,27 +912,15 @@ def generate_token(self, batches: List[CausalLMBatch]) -> Tuple[List[Generation]
                 else:
                     generated_text = None
 
-                # Prefill
-                if stopping_criteria.current_tokens == 1 and request.prefill_logprobs:
-                    # Remove generated token to only have prefill and add nan for first prompt token
-                    prefill_logprobs = [float("nan")] + next_token_logprobs
-                    prefill_token_ids = all_input_ids[0: new_input_length - 1]
-                    prefill_texts = self.tokenizer.batch_decode(
-                        prefill_token_ids,
-                        clean_up_tokenization_spaces=False,
-                        skip_special_tokens=False,
-                    )
-                    prefill_tokens = PrefillTokens(prefill_token_ids, prefill_logprobs, prefill_texts)
-                else:
-                    prefill_tokens = None
-
+                assert request.prefill_logprobs == False
```

Comment:
In the case of greedy we will always fall into the statement "else". I don't think it's that expensive to leave the original code.